src/data-workspace/display/table-custom-data-set.js

```diff
@@ -0,0 +1,108 @@
+import {
+    DataTable,
+    TableHead,
+    DataTableRow,
+    DataTableColumnHeader,
+    TableBody,
+    DataTableCell,
+} from '@dhis2/ui'
+import DOMPurify from 'dompurify'
+import PropTypes from 'prop-types'
+import React from 'react'
+import styles from './table.module.css'
+
+// Needs to have the same width as the table, so can't use the one from
+// @dhis2/ui
+const DataTableToolbar = ({ children, columns }) => (
+    <tr>
+        <th className={styles.titleCell} colSpan={columns.toString()}>
+            {children}
+        </th>
+    </tr>
+)
+
+DataTableToolbar.propTypes = {
+    children: PropTypes.any.isRequired,
+    columns: PropTypes.number.isRequired,
+}
+
+const TableCustomDataSet = ({ title, columns, rows }) => (
+    <>
+        <DataTable className={styles.dataTable} width="auto">
+            <TableHead>
+                <DataTableToolbar columns={columns.length}>
+                    <span
+                        dangerouslySetInnerHTML={{
+                            __html: DOMPurify.sanitize(title),
```

Comment:
this seems like the title should not be HTML? Isn't this just the dataset name, and so if someone calls their dataset `<h2>My data set</h2>`, I guess they should get that string literal back? Not parsed HTML?